src/utils/identifiers.ts

```diff
@@ -33,4 +34,21 @@ export const solanaId = (address: string): string => {
 export const solanaTrackId = (address: string, id: string): string => {
   const suffix = id !== '' ? `/${id}` : '';
   return solanaId(address) + suffix;
-}
\ No newline at end of file
+}
+
+export const controlledEthereumAddressFromId = (id: string | undefined): string | undefined => {
+  if (id === undefined) { return; }
+
+  if (isControlledEthereumAddress(id)) {
+    return id
+  }
+
+  const parts = id.split('/');
+  if (parts[1] && isControlledEthereumAddress(parts[1])) {
+    return parts[1];
+  }
+}
+
+const isControlledEthereumAddress = (address: string): boolean => {
```

Comment:
can we just use https://docs.ethers.io/v5/api/utils/address/#utils-isAddress ?

Reply:
Couldn't find a good function but I was looking in web3js. That from ethers should work.

Reply:
I wasn't 100% confident that burn addresses are inaccessible wallets. I can remove though.